### ISSUE TITLE
upgrade from 1.4.4 to 1.4.5 failed

### ISSUE THREAD

**Issue by rallek** [zikula/core#3340]
| Q                 | A
| ----------------- | ---
| Zikula Version    | 1.4.5
| PHP Version       | 5.5.30

### Expected behavior
Upgrade would run as usual

### Actual behavior
Upgrade went nice until click on finish. The next is having this failure:

````
[1] Symfony\Component\Routing\Exception\RouteNotFoundException: Unable to generate a URL for the named route "rkwebsitehelpermodule_admin_index" as such route does not exist.
    at n/a
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/app/cache/prod/appProdDebugProjectContainerUrlGenerator.php line 367

    at appProdDebugProjectContainerUrlGenerator->generate('rkwebsitehelpermodule_admin_index', array(), '1')
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/vendor/jms/i18n-routing-bundle/JMS/I18nRoutingBundle/Router/I18nRouter.php line 144

    at JMS\I18nRoutingBundle\Router\I18nRouter->generate('rkwebsitehelpermodule_admin_index')
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/system/AdminModule/Controller/AdminController.php line 440

    at Zikula\AdminModule\Controller\AdminController->categorymenuAction('1')
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/system/AdminModule/Controller/AdminController.php line 274

    at Zikula\AdminModule\Controller\AdminController->adminpanelAction(null)
        in  line 

    at call_user_func_array(array(object(AdminController), 'adminpanelAction'), array(null))
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php line 144

    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Zikula_Request_Http), '1')
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php line 64

    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Zikula_Request_Http), '1', true)
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/ContainerAwareHttpKernel.php line 69

    at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(object(Zikula_Request_Http), '1', true)
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php line 185

    at Symfony\Component\HttpKernel\Kernel->handle(object(Zikula_Request_Http))
        in /home/www/www.oldtimer-ig-osnabrueck.de/Zikula_Core-1.4.5-test/index.php line 23
````

### Steps to reproduce
Follow the upgrade instruction for the normal way.

in detail:
- rescue the both config files, the theme, the folder userdata and the modules
- download an unpack the build 101 with ``tar -xpzf ...``
- place back both config files, the theme and userdata, not modules
- go to your website and follow the upgrade procedure

I can go to the routes module and can rebuild the routes. After that this error has gone.

**Comment by craigh** [comment]
Do you have a hardcoded route in your theme? `rkwebsitehelpermodule_admin_index`

**Comment by rallek** [comment]
No. I do not have. The websitehelper is a MOST module. As far as I was able to see when I opened the admin sidebar all modern modules do have the missing route as a comment beside. Unfortunately I did not make a screen shoot of it.

**Comment by rallek** [comment]
It might be the issue is not new. The only modern module I upgraded before was MediaModule. There I can not remember this issue. MOST modules I do nat have had before in my upgrades from 1.4.3 to 1.4.4.

It might be also a MOST issue. I do not know. Therefore I did not vote yet.

**Comment by craigh** [comment]
is the code for this module online anywhere?

can you search for `rkwebsitehelpermodule_admin_index` and tell me where it is used?

**Comment by rallek** [comment]
here I have it in github:
https://github.com/rallek/oldtimer-ig-osnabrueck.de/tree/master/generatedModules/WebsitehelperModule

**Comment by rallek** [comment]
at two places:
https://github.com/rallek/oldtimer-ig-osnabrueck.de/search?utf8=%E2%9C%93&q=rkwebsitehelpermodule_admin_index

**Comment by craigh** [comment]
nice. 👍 

This sounds similar to some other issues we've seen but not sure. I'm assuming it is not a new issue, but just something not fixed from past or something not core.

I'll wait for @Guite 's opinions and then we can proceed.

> I can go to the routes module and can rebuild the routes. After that this error has gone.

since there is a workaround, I don't see it as a blocker.

Also - "rebuilding the routes" is the same as *deleting the cache*. FYI

**Comment by Guite** [comment]
> since there is a workaround, I don't see it as a blocker.

Yep.

**Comment by rallek** [comment]
see my detaild explanation in #3345 
But it is no blocker, that is true

**Comment by Guite** [comment]
schedule to 1.4.6... you might restest this after #3224 and #2684 have been done

**Comment by craigh** [comment]
note to self:

the current instructions in the upgrade process say that modules should be removed, run the upgrade, then return modules one at a time.

this obviously means that the modules are not present and active when rebuilding the route cache and this causes the errors.

the question is therefore how to modify the upgrade process or handle routes in this situation.

**Comment by Guite** [comment]
One possibility would be to simply catch RouteNotFoundException and add a flash like "now readd your modules". But this would require that the system knows that it just has been upgraded.

**Comment by rallek** [comment]
I upgraded including the modules from 1.4.4 to 1.4.5 my live site. There I did not have any problems. That looks quite good!

Might be we do not need to add modules one by one after upgrading inside 1.4.x?

**Comment by craigh** [comment]
> Might be we do not need to add modules one by one after upgrading inside 1.4.x?

This was my thought as well.

**Comment by craigh** [comment]
@rallek now that the locale stuff has been added, will you please retest this when you have a few minutes? thanks.

**Comment by rallek** [comment]
will do latest during next weekend

**Comment by rallek** [comment]
Anything I should be aware? E.g. upgrade with/without modules?

**Comment by craigh** [comment]
try it _with_

**Comment by rallek** [comment]
I tried it with _everything_ :-)

My preparatioin before upgrading:

- backup of my database
- create dir ``144bak`` and copy all needed files in there
- copy module/ 
- delete 144bak/modules/zikula/
- copy userdata
- copy app/config/custom_parameters.yml
- copy config/personal_config.php
- copy theme/myTheme/
- copy app/Resources/translation/
- copy app/Resources/locale/

The content of app/Resources/ can be seen here: https://github.com/rallek/oldtimer-ig-osnabrueck.de/tree/master/patch/app/Resources

- Next I downloaded 1.4.6 build 68 from CI and unpacked it in a folder Zikula_Core-1.4.6
- 
- Now I copied my 144bak into the Zikula folder
- 
- At the end I changed the url for my host to the new zikula folder and want to start my upgrade. But this failed.

Next I will try without my translation files. Might be, they are the reason.


Here my stacktrace:
````
[1] Symfony\Component\Routing\Exception\MissingMandatoryParametersException: Some mandatory parameters are missing ("returnUrl") to generate a URL for route "de__RG__zikulausersmodule_access_login".
    at n/a
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/Routing/Generator/UrlGenerator.php line 165

    at Symfony\Component\Routing\Generator\UrlGenerator->doGenerate(array('returnUrl'), array('_controller' => 'Zikula\UsersModule\Controller\AccessController::loginAction', '_zkBundle' => 'ZikulaUsersModule', '_zkModule' => 'ZikulaUsersModule', '_zkType' => 'access', '_zkFunc' => 'login', '_locale' => 'de'), array(), array(array('variable', '/', '[^/]++', 'returnUrl'), array('text', '/login')), array('authenticationMethod' => 'native_uname'), 'de__RG__zikulausersmodule_access_login', '1', array(), array())
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/app/cache/prod/appProdDebugProjectContainerUrlGenerator.php line 1021

    at appProdDebugProjectContainerUrlGenerator->generate('de__RG__zikulausersmodule_access_login', array('authenticationMethod' => 'native_uname'), '1')
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/jms/i18n-routing-bundle/JMS/I18nRoutingBundle/Router/I18nRouter.php line 128

    at JMS\I18nRoutingBundle\Router\I18nRouter->generate('zikulausersmodule_access_login', array('authenticationMethod' => 'native_uname'))
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/lib/Zikula/Bundle/CoreBundle/EventListener/SiteOffListener.php line 103

    at Zikula\Bundle\CoreBundle\EventListener\SiteOffListener->onKernelRequestSiteOff(object(GetResponseEvent), 'kernel.request', object(Zikula_EventManager))
        in  line 

    at call_user_func(array(object(SiteOffListener), 'onKernelRequestSiteOff'), object(GetResponseEvent), 'kernel.request', object(Zikula_EventManager))
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php line 184

    at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(array(object(DebugHandlersListener), 'configure'), array(object(OutputCompressionListener), 'onKernelRequest'), array(object(StripFrontControllerListener), 'onKernelRequest'), array(object(ValidateRequestListener), 'onKernelRequest'), array(object(CollapseableBlockListener), 'addCollapseableBehavior'), array(object(LocalizedVariableListener), 'onKernelRequest'), array(object(SessionListener), 'onKernelRequest'), array(object(SiteOffListener), 'onKernelRequestSiteOff'), array(object(FilterListener), 'idsInputFilter'), array(object(FragmentListener), 'onKernelRequest'), array(object(FrontControllerListener), 'pageLoad'), array(object(RouterListener), 'onKernelRequest'), array(object(SessionExpireListener), 'onKernelRequestSessionExpire'), array(object(LocaleListener), 'onKernelRequest'), array(object(LocaleListener), 'onKernelRequest'), array(object(TranslatorListener), 'onKernelRequest'), array(object(Firewall), 'onKernelRequest'), array(object(KernelListener), 'onRequest'), array(object(KernelListener), 'onRequest'), array(object(KernelListener), 'onRequest'), array(object(DefaultPageAssetSetterListener), 'setDefaultPageAssets'), array(object(DefaultPageVarSetterListener), 'setDefaultPageVars'), array(object(VoterInitializerListener), 'onKernelRequest'), array(object(LocaleListener), 'onKernelRequest'), array(object(LoggerListener), 'onKernelRequest'), array(object(BlameListener), 'onKernelRequest')), 'kernel.request', object(GetResponseEvent))
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php line 46

    at Symfony\Component\EventDispatcher\EventDispatcher->dispatch('kernel.request', object(GetResponseEvent), null)
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/lib/legacy/Zikula/EventManager.php line 103

    at Zikula_EventManager->dispatch('kernel.request', object(GetResponseEvent))
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php line 125

    at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Zikula_Request_Http), '1')
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php line 64

    at Symfony\Component\HttpKernel\HttpKernel->handle(object(Zikula_Request_Http), '1', true)
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/DependencyInjection/ContainerAwareHttpKernel.php line 69

    at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle(object(Zikula_Request_Http), '1', true)
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php line 185

    at Symfony\Component\HttpKernel\Kernel->handle(object(Zikula_Request_Http))
        in /home/www/www.oldtimer-ig-osnabrueck.de/test/Zikula_Core-1.4.6/index.php line 23
````

**Comment by rallek** [comment]
Now it was running well but other problems are coming up. I will create new tickets.

**Comment by rallek** [comment]
Is it by purpose all my modules are set to inactive?

**Comment by craigh** [comment]
please test **_without_** a custom theme.

**Comment by Guite** [comment]
> Next I will try without my translation files. Might be, they are the reason.
> Here my stacktrace:

Yes, this is the obsolete translation of the login route I've shown you recently.

**Comment by Guite** [comment]
To close this issue I propose changing the update instructions to keeping modules in place instead of temporarily removing them. This way I successfully updated two installations from 1.4.4 to 1.4.5.

Opinions?

**Comment by rallek** [comment]
Thats fine for me
